coderd/notifications/enqueuer.go

```diff
@@ -92,6 +101,13 @@ func (s *StoreEnqueuer) Enqueue(ctx context.Context, userID, templateID uuid.UUI
 			return nil, ErrCannotEnqueueDisabledNotification
 		}
 
+		// If the enqueue fails due to a dedupe hash conflict, this means that a notification has already been enqueued
+		// today with identical properties. It's far simpler to prevent duplicate sends in this central manner, rather than
+		// having each notification enqueue handle its own logic.
+		if database.IsUniqueViolation(err, database.UniqueNotificationMessagesDedupeHashIndex) {
+			return nil, ErrDuplicate
```

Comment:
Do we actually care if the dedupe happens from the outside? Is it simpler for callers to just log and return `nil, nil`?